### DISCUSSION TITLE
Re-Implement the ForceFeedback SCtrl

### DISCUSSION THREAD

**Discussion by SuperFromND** [ikemen-engine/Ikemen-GO#1549] [Feature Request] [answered]
This is more of a reminder than a proper discussion, but as of right now Ikemen GO does not yet support the state controller `ForceFeedback`, which allows the controller to rumble for a given duration and amplitude. 

We do sort-of implement this already (see [bytecode.go at line ~7236 as of this writing](https://github.com/ikemen-engine/Ikemen-GO/blob/acabcf802d9aede0a91aaa9802a577e887617b9c/src/bytecode.go#L7236C33-L7236C33)), but it seems like the GLFW input library itself doesn't give any way for us to actually *use* this data; i.e. sending it to a controller, so instead the implementation is commented out and the controller just acts as a no-op.

GLFW has had an open ticket for implementing general-purpose rumble functions (https://github.com/glfw/glfw/issues/57) open since 2013, but progress doesn't seem to have been made much since around 2020. Kinc *does* appear to have an API command for it, [`kinc_gamepad_rumble()`](https://kinc.tech/api/gamepad_8h.html#a3d52cdbcbfa52e09910f37fdb05d584a), but I'm not familiar enough with the library (nor how Ikemen uses it) to implement it myself.

**Comment by potsmugen** [comment]
This should be implemented if nothing else because it was always there in Mugen but never worked, if I recall correctly. Just something that has to be done.

**Reply by SuperFromND** [reply]
Agreed, it's one of those things that I think would be nice to finally "restore" for lack of better term.

**Comment by shinra358** [comment]
yes please

**Comment by K4thos** [comment, marked as answer]
already implemented
